internal/grpc/services/gateway/usershareprovider.go

```diff
@@ -78,18 +89,33 @@ func (s *svc) CreateShare(ctx context.Context, req *collaboration.CreateShareReq
 					Status: denyGrantStatus,
 				}, err
 			}
-			return res, nil
+			// return res, nil
+		} else {
+			addGrantStatus, err := s.addGrant(ctx, req.ResourceInfo.Id, req.Grant.Grantee, req.Grant.Permissions.Permissions)
+			if err != nil {
+				return nil, errors.Wrap(err, "gateway: error adding grant to storage")
+			}
+			if addGrantStatus.Code != rpc.Code_CODE_OK {
+				return &collaboration.CreateShareResponse{
+					Status: addGrantStatus,
+				}, err
+			}
 		}
+	}
 
-		addGrantStatus, err := s.addGrant(ctx, req.ResourceInfo.Id, req.Grant.Grantee, req.Grant.Permissions.Permissions)
-		if err != nil {
-			return nil, errors.Wrap(err, "gateway: error adding grant to storage")
-		}
-		if addGrantStatus.Code != rpc.Code_CODE_OK {
-			return &collaboration.CreateShareResponse{
-				Status: addGrantStatus,
-			}, err
-		}
+	// Then we commit to the db
+	// ------------------------
+
+	// TODO the user share manager needs to be able to decide if the current user is allowed to create that share (and not eg. incerase permissions)
```

Comment:
This comment comes from before and would apply to the storage provider rather than the DB as "share manager". But as a matter of fact, it does not apply to us in the sense that our storage drivers - EOS in particular here, and CephFS in the future - are doing that check. So it can be removed.